react/Infos/index.jsx

```diff
@@ -1,73 +1,119 @@
 import React from 'react'
 import cx from 'classnames'
 import PropTypes from 'prop-types'
-import styles from './styles.styl'
-import Icon, { iconPropType } from '../Icon'
+import Icon from '../Icon'
 import Text, { SubTitle } from '../Text'
+import Stack from '../Stack'
+import palette from '../palette'
+import createDepreciationLogger from '../helpers/createDepreciationLogger'
+
+import styles from './styles.styl'
 
-const Infos = ({ actionButton, icon, isImportant, text, className, title }) => {
+export const Infos = ({
+  description,
+  action,
+  dismissAction,
+  theme,
+  className
+}) => {
   return (
-    <div
-      className={cx(
-        styles['Infos'],
-        'u-flex',
-        'u-flex-column',
-        'u-p-1',
-        'u-mih-2',
-        'u-flex-justify-center',
-        'u-ta-left',
-        'u-stack-m',
-        isImportant ? 'u-bg-chablis' : 'u-bg-paleGrey',
-        className
+    <div className={cx(styles['Infos'], styles[`Infos--${theme}`], className)}>
+      <Stack spacing="m">
+        <div className={styles['Infos-description']}>
+          <Stack spacing="s">{description}</Stack>
+        </div>
+        {action && <div>{action}</div>}
+      </Stack>
+      {dismissAction && (
+        <button className={styles['Info-close']} onClick={dismissAction}>
+          <Icon icon="cross" color={palette['coolGrey']} />
+        </button>
       )}
-    >
-      {!!title && (
-        <SubTitle
-          className={cx({
-            'u-pomegranate': isImportant
-          })}
-        >
-          {title}
-        </SubTitle>
-      )}
-      <div className={cx('u-flex', 'u-w-100')}>
-        {icon && (
-          <Icon
-            icon={icon}
-            className={cx(
-              styles['Infos__icon'],
-              'u-w-1',
-              'u-h-1',
-              'u-flex-shrink-0'
-            )}
-          />
-        )}
-        <Text
-          className={cx('u-ta-left', {
-            ['u-pl-half']: icon !== null
-          })}
-        >
-          {text}
-        </Text>
-      </div>
-      {!!actionButton && <div className="u-flex-shrink-0">{actionButton}</div>}
     </div>
   )
 }
 
-Infos.defaultProps = {
-  icon: null
+Infos.propTypes = {
+  /** The textual part of the info, including titles */
+  description: PropTypes.element.isRequired,
+  /** One or more call to actions */
+  action: PropTypes.element,
+  /** If supplied, displays a cross in the top right corner that calls the prop when clicked */
+  dismissAction: PropTypes.func,
+  /** Extra classnames to apply to the root element */
+  className: PropTypes.string,
+  /** Controls the background color of the component */
+  theme: PropTypes.oneOf(['primary', 'secondary', 'danger'])
```

Comment:
I have my doubt about the name of this prop. In effect it really only controls the background color at the moment, though it could do more in the future..? It could impact the text color or the cross for example. And `theme`is coherent with the `Button` API. That said I could also add a default css color and leave the implementer to define the color they need. Any thoughts?